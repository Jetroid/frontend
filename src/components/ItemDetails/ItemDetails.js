import React from "react";
import { Link } from "react-router-dom";
import { Container, Row, Col } from "reactstrap";
import RelatedContent from "../RelatedContent";
import Gallery from "../Gallery";
import ReactPlayer from 'react-player'
import BookmarkToggle from "../BookmarkToggle";
import FloatingActionButton from "material-ui/FloatingActionButton";
import ContentPencil from "material-ui/svg-icons/image/edit";
import caseIconSettings from "../../img/pp-case-icon-settings.svg";
import caseIconFB from "../../img/pp-case-icon-fb.svg";
import caseIconTW from "../../img/pp-case-icon-tw.svg";
import caseIconShare from "../../img/pp-case-icon-share.svg";
import { ShareButtons } from "react-share";
import htmlToText from "html-to-text";
import "./ItemDetails.css";
import TimeAgo from "react-timeago";

export default class ItemDetails extends React.Component {
  render() {
    const { FacebookShareButton, TwitterShareButton } = ShareButtons;
    const isAuthenticated = this.props.isAuthenticated;
    let intl = this.props.intl;
    let thing = this.props.data;
    let bookmarked = isAuthenticated && thing.bookmarked;
    let bookmarkIcon = <div />;
    if (isAuthenticated) {
      bookmarkIcon = (
        <BookmarkToggle
          thingType="case"
          thingID={thing.id}
          bookmarked={bookmarked}
        />
      );
    }
    let bodyText = htmlToText.fromString(thing.body);
    let currentUrl = "https://participedia.xyz" +  this.props.location.pathname;
    let textFacebook = bodyText.substring(0, 240) + "...";

    let lead;
    let awsUrl = process.env.REACT_APP_ASSETS_URL;
    if (thing && thing.lead_image) {
      lead = awsUrl + encodeURIComponent(thing.lead_image.url);
    }

    let first_author = thing.authors[0];
    let first_author_url = "/users/" + first_author.user_id;
    let first_author_name = first_author.name;
    let last_author = thing.authors.slice(-1)[0];
    let last_author_name = last_author.name;
    let last_author_url = "/users/" + last_author.user_id;
    let id = this.props.id;
    let type = thing.type;
    let editLinkUrl = `/${type}/${id}/edit`;
    let detailedBits = React.createElement(this.props.details, {
      case: thing,
      intl
    });

    // XXX L10N -- lots of strings to extract.
    return (
      <div>
        <div className="main-contents">
          <Container className="detailed-case-component" fluid={true}>
            <Row>
              <Col md="3" className="hidden-sm-down sidepanel hidden-sm-down">
                {detailedBits}
                <RelatedContent thing={thing} intl={intl} />
              </Col>
              <Col md="8" xs="12" className="main-area">
                <div className="case-box">
                  <h2 className="category">
                    {intl.formatMessage({ id: thing.type })}
                  </h2>
                  <h2 className="case-title">
                    {thing.title}
                  </h2>
                  { thing.vidURL ?
                  <Row>
                    <Col className="vid-container" md="6"><ReactPlayer width="100%"  controls={true} url={thing.vidURL} /></Col>
                    <Col md="6"><Gallery thing={thing} /></Col>
                  </Row>
                  :
                  <Gallery thing={thing} />
                  }
                  <div className="authorship-details">
                    <p className="author-line">
                      First submitted by&nbsp;
                      <Link to={first_author_url}>
                        {first_author_name}
                      </Link>
                    </p>
                    <p className="date-line">
                      <TimeAgo date={thing.post_date} />
                    </p>
                    <p className="author-line">
                      Most recent changes by&nbsp;
                      <Link to={last_author_url}>
                        {last_author_name}
                      </Link>
                    </p>
                    <p className="date-line">
                      <TimeAgo date={thing.updated_date} />
                    </p>
                  </div>
                  <div
                    className="case-html"
                    dangerouslySetInnerHTML={{ __html: thing.body }}
                  />
                </div>
              </Col>
              <Col md="1" className="case-tools hidden-sm-down">
                <div className="top-icons">
                  {bookmarkIcon}
                  <a href="#"><img src={caseIconSettings} alt="" /></a>
                  <FacebookShareButton
                    url={currentUrl}
                    title={thing.title}
                    description={textFacebook}
                    picture={lead}
                  >
                    <img src={caseIconFB} alt="" />
                  </FacebookShareButton>
                  <TwitterShareButton
                    url={currentUrl}
                    title={thing.title}
                  >
                    <img src={caseIconTW} alt="" />
                  </TwitterShareButton>
                  <a href="#"><img src={caseIconShare} alt="" /></a>
                </div>
              </Col>
            </Row>
          </Container>
          <Link to={editLinkUrl}>
            <FloatingActionButton className="editButton">
              <ContentPencil />
            </FloatingActionButton>
          </Link>
        </div>
      </div>
    );
  }
}
