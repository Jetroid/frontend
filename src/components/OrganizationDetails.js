import React from "react";

import SearchLink from "../components/SearchLink";
import CountryMap from "../components/CountryMap";
import LinkedPropertyGroupWithHeading from "../components/LinkedPropertyGroupWithHeading";
import ListWithHeading from "../components/ListWithHeading";

export default class OrganizationDetails extends React.Component {
  render() {
    let thing = this.props.case;
    let intl = this.props.intl;
    let tags = <div />;
    if (thing.tags) {
      tags = thing.tags.map(tag =>
        <SearchLink intl={intl} key={tag} tag="tag" value={tag} />
      );
    }
    let ed = null;
    if (thing.executive_director) {
      ed = (
        <div>
          <p className="sub-heading">
            {intl.formatMessage({ id: "executive_director" })}
          </p>
          <div className="sub-sub-heading">
            {thing.executive_director}
          </div>
        </div>
      );
    }
    return (
      <div>
        <CountryMap
          city={thing.location.city}
          countrycode={thing.location.country}
        />
        <p className="d-none d-sm-block d-md-block d-lg-block d-xl-block sub-heading">
          {intl.formatMessage({ id: "data" })}
        </p>
        <p className="sub-sub-heading">
          {intl.formatMessage({ id: "tags_title" })}
        </p>
        <div className="tags">
          {tags}
        </div>
        <ListWithHeading
          intl={intl}
          heading="links"
          property="links"
          thing={thing}
        />
        {ed}

        <LinkedPropertyGroupWithHeading
          intl={intl}
          heading="sector"
          property="sector"
          thing={thing}
        />
      </div>
    );
  }
}
