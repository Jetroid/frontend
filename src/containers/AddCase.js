import CaseForm from '../components/CaseForm'
import { reduxForm } from 'redux-form'

const mapStateToProps = (state /* , ownProps */) => {
  return state.form
}

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    onSubmit: (data) => {
      console.log('FORM PROPS', data, ownProps)
    }
  }
}

export const fields = [ 'firstName', 'lastName', 'age', 'color', 'bio' ]

let AddCaseForm = reduxForm({
  form: 'ncase',
  fields: ['firstName', 'lastName', 'email']
}, mapStateToProps, mapDispatchToProps)(CaseForm)

export default AddCaseForm
