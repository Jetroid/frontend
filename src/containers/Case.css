.main-contents {
  min-height: 200px;
}

.editButton {
  position: fixed;
  bottom: 2em;
  right: 2em;
}



.detailed-case-component {
  margin: 0 auto;
  overflow: hidden;
  padding-bottom: 32px;
  padding-top: 64px;
}

.detailed-case-component .sidebar {
  padding: 0 32px;
  width: 200px;
  float: left;
  margin-top: 48px;
}

.detailed-case-component .sidebar .case-map {
  width: 100%;
}

.detailed-case-component .sidebar .case-location {
  font-weight: 500;
  color: #3f51b2;
  margin-top: 16px;
  margin-bottom: 16px;
}

.detailed-case-component .sidebar .progress-bar {
  width: 100%;
  background-color: #ddd;
  height: 16px;
}

.detailed-case-component .sidebar .progress-bar .progress-fill {
  height: 16px;
  background-color: #3f51b2;
}

.detailed-case-component .sidebar .progress-complete {
  font-size: 14px;
  margin-top: 8px;
  font-weight: 400;
}

.detailed-case-component .sidebar .sub-heading {
  font-size: 32px;
  font-weight: 400;
  line-height: 40px;
  margin-bottom: 12px;
  margin-top: 16px;
}

.detailed-case-component .sidebar .sub-sub-heading {
  font-size: 14px;
  font-weight: 500;
  color: #777;
  line-height: 18px;
  margin-bottom: 8px;
  margin-top: 16px;
}

.detailed-case-component .sidebar .tags a {
  display: block;
  font-size: 14px;
  color: #888;
  font-weight: 400;
  line-height: 18px;
  margin-bottom: 4px;
}

.detailed-case-component .sidebar .related-content a {
  font-size: 14px;
  font-weight: 500;
  color: #777;
  line-height: 18px;
  margin-bottom: 8px;
  margin-top: 8px;
  position: relative;
  display: block;
  padding-left: 32px;
}

.detailed-case-component .sidebar .related-content a:before {
  content: "";
  background-image: url("/img/arrow-down-icon.png");
  display: block;
  position: absolute;
  left: 0;
  width: 24px;
  height: 24px;
  opacity: 0.5;
  margin-top: -4px;
}

.detailed-case-component .main-area {
  /*float: right;*/
  width: 1248px;
  padding-right: 32px;
}

.detailed-case-component .main-area .case-box {
  width: 960px;
  float: left;
  padding: 32px 0;
}

.detailed-case-component .main-area .case-box .category {
  color: #ff6f00;
  font-size: 48px;
  line-height: 60px;
  font-weight: 500;
  margin-bottom: 8px;
}

.detailed-case-component .main-area .case-box .case-title {
  color: #3f51b2;
  font-size: 48px;
  line-height: 60px;
  font-weight: 500;
  margin-bottom: 48px;
  width: 780px;
}

.detailed-case-component .main-area .case-box .case-images {
  overflow: hidden;
}

.detailed-case-component .main-area .case-box .case-images .thumbnail {
  width: 288px;
  height: 160px;
  background-size: cover;
  background-position: center;
  margin-bottom: 10px;
  background-color: #eee;
  float: left;
  margin-right: 32px;
}

.detailed-case-component .main-area .case-box .authorship-details {
  margin-top: 32px;
  margin-bottom: 32px;
}

.detailed-case-component .main-area .case-box .authorship-details .author-line {
  font-size: 14px;
  font-weight: 500;
  color: #777;
  line-height: 18px;
}

.detailed-case-component .main-area .case-box .authorship-details .author-line a {
  color: #ff6f00;
  font-weight: 500;
}

.detailed-case-component .main-area .case-box .authorship-details .date-line {
  font-size: 14px;
  color: #777;
  font-weight: 400;
  line-height: 18px;
  margin-bottom: 6px;
}

.detailed-case-component .main-area .case-box .case-html h3 {
  font-size: 32px;
  font-weight: 400;
  line-height: 40px;
  margin-bottom: 12px;
  margin-top: 32px;
}

.detailed-case-component .main-area .case-box .case-html p,
.detailed-case-component .main-area .case-box .case-html li {
  font-size: 16px; /* XXX */
  font-weight: 100;
  line-height: 24px;
  margin-bottom: 16px;
}

.detailed-case-component .main-area .case-box .case-html h2 {
  font-size: 20px;
  font-weight: 400;
  line-height: 24px;
  margin-bottom: 16px;
}

.detailed-case-component .main-area .case-box .case-html li {
  list-style-type: square;
  margin-left: 32px;
  margin-bottom: 8px;
  padding-left: 8px;
}

.detailed-case-component .main-area .case-box .case-html ol li {
  list-style-type: lower-roman;
}

.detailed-case-component .main-area .case-box .case-html a {
  color: #ff6f00;
  font-weight: 400;
}

.detailed-case-component .main-area .case-tools {
  position: fixed;
  right: 6em;
  top: 4em;
  z-index: 12000;
  /*float: right;
  margin-right: 32px;
  padding-right: 32px;*/
}

.detailed-case-component .main-area .case-tools .top-icons {
  margin-top: 40px;
  position: fixed;
}

.detailed-case-component .main-area .case-tools .top-icons a {
  display: block;
  margin-bottom: 16px;
}

.detailed-case-component .main-area .case-tools .top-icons a img {
  width: 40px;
}

.detailed-case-component .main-area .case-tools .top-icons a:first-child img {
  width: 32px;
  margin: 4px;
}

.detailed-case-component .main-area .case-tools .bottom-icons {
  position: absolute;
  bottom: 0;
  margin-bottom: 48px;
  position: fixed;
}

.detailed-case-component .main-area .case-tools .bottom-icons a {
  position: relative;
  display: block;
}

.detailed-case-component .main-area .case-tools .bottom-icons a img {
  width: 80px;
  margin-left: -20px;
}

.detailed-case-component .main-area .case-tools .bottom-icons a::before {
  content: "";
  background-color: #ff6f00;
  position: absolute;
  color: #fff;
  font-weight: 400;
  left: 65px;
  top: 19px;
  width: 12px;
  height: 12px;
  transform: rotate(45deg);
  display: none;
}

.detailed-case-component .main-area .case-tools .bottom-icons a::after {
  content: "Edit";
  background-color: #ff6f00;
  padding: 6px 12px;
  position: absolute;
  display: none;
  color: #fff;
  font-weight: 400;
  left: 70px;
  top: 10px;
}

.detailed-case-component .main-area .case-tools .bottom-icons a:hover::before,
.detailed-case-component .main-area .case-tools .bottom-icons a:hover::after {
  display: block;
}
