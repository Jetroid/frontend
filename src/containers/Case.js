import React from "react";
import { bool, object } from "prop-types";
import { connect } from "react-redux";
import { injectIntl, intlShape } from "react-intl";
import api from "../utils/api";
import CaseDetails from "../components/CaseDetails";
import ItemFetcher from "./ItemFetcher";

function mapStateToProps({ auth }) {
  const { isAuthenticated } = auth;

  return {
    isAuthenticated
  };
}

export class Case extends React.Component {
  toggleFeatured(thing, featured) {
    console.log("setting featured for ", thing, " to ", featured);
    // XXX needs API to exist.
  }
  render() {
    return (
      <ItemFetcher
        api={api.fetchCaseById}
        id={Number(this.props.match.params.nodeID)}
        details={CaseDetails}
        toggleFeatured={this.toggleFeatured.bind(this)}
        {...this.props}
      />
    );
  }
}
Case.propTypes = {
  intl: intlShape.isRequired,
  isAuthenticated: bool,
  location: object
};

export default connect(mapStateToProps)(injectIntl(Case));
