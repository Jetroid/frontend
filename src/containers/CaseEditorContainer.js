import React, { Component } from "react";
import CaseEditor from "../components/CaseEditor/CaseEditor";
import api from "../utils/api";

// May make sense to revert back to redux for this work. this is duplicative.
// Merge with EditCase, which is similar but not currently being used.

export default class CaseEditorContainer extends Component {
  getNouns() {
    let component = this;
    let nouns = {
      CASE: "cases",
      ORGANIZATION: "organizations",
      METHOD: "methods"
    };
    Object.keys(nouns).map((k, i) =>
      api.fetchNouns(k).then(function(the_nouns) {
        component.setState({ [nouns[k]]: the_nouns });
      }));
  }

  componentWillMount() {
    let component = this;
    api.fetchCaseById(this.props.params.nodeID).then(function(thing) {
      if (thing) {
        thing.start_date = new Date(thing.start_date);
        thing.end_date = new Date(thing.end_date);
        thing.updated_date = new Date(thing.updated_date);
      }
      component.setState({ case: thing });
    });
    this.getNouns();
  }

  onSubmit(values, state, props, instance) {
    let router = this.props.router;
    let location = this.props.location;
    api.saveCase(state).then(function(thing) {
      let pathparts = location.pathname.split("/");
      pathparts.pop();
      router.push(pathparts.join("/"));
    });
  }

  render() {
    if (
      this.state &&
      this.state.case &&
      this.state.cases &&
      this.state.methods &&
      this.state.organizations
    ) {
      // need to turn this into a loop
      let casesArr = Object.keys(this.state.cases).map(k => k);
      let methodsArr = Object.keys(this.state.methods).map(k => k);
      let orgsArr = Object.keys(this.state.organizations).map(k => k);
      return (
        <CaseEditor
          cases={casesArr}
          methods={methodsArr}
          organizations={orgsArr}
          thing={this.state.case}
          onSubmit={this.onSubmit.bind(this)}
        />
      );
    } else {
      return <div />;
    }
  }
}
