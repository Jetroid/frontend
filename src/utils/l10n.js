import locales from "../../public/locales.json";

export function getBestMatchingMessages(locale) {
  if (locale in locales) {
    return locales[locale];
  }
  let [language] = locale.split("-");
  if (language in locales) {
    return locales[language];
  }
  return locales["en"];
}

export function getFirstBrowserLanguage() {
  let nav = window.navigator;
  let browserLanguagePropertyKeys = [
    "language",
    "browserLanguage",
    "systemLanguage",
    "userLanguage"
  ];
  let i = 0;
  let language = "";

  // support for HTML 5.1 "navigator.languages"
  if (Array.isArray(nav.languages)) {
    for (let i = 0; i < nav.languages.length; i++) {
      let language = nav.languages[i];
      if (language && language.length) {
        return language;
      }
    }
  }

  // support for other well known properties in browsers
  for (i = 0; i < browserLanguagePropertyKeys.length; i++) {
    language = nav[browserLanguagePropertyKeys[i]];
    if (language && language.length) {
      return language;
    }
  }

  return null;
}
