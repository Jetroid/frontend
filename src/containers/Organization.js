import React from 'react'
import './Case/Case.css'
import {injectIntl, intlShape} from 'react-intl'
import api from '../utils/api'
import moment from 'moment'
import { Container, Col } from 'reactstrap'
import CountryMap from '../components/CountryMap'
import FloatingActionButton from 'material-ui/FloatingActionButton'
import ContentPencil from 'material-ui/svg-icons/image/edit'
import caseIconBookmark from '../img/pp-case-icon-bookmark.png'
import caseIconSettings from '../img/pp-case-icon-settings.png'
import caseIconFB from '../img/pp-case-icon-fb.png'
import caseIconTW from '../img/pp-case-icon-tw.png'
import caseIconShare from '../img/pp-case-icon-share.png'


export class Organization extends React.Component {
  componentWillMount () {
    let component = this
    api.fetchOrgById(this.props.params.nodeID).then(function (data) {
      component.setState({data: data, htmlBody: data.body})
    })
  }

  getInnerHTML () {
    let input = ''
    if (this.state && this.state.htmlBody) {
      input = this.state.htmlBody
    }
    return {__html: input}
  }

  render () {
    if (this.state && this.state.data) {
      let organizationObject = this.state.data
      let tags, communication_modes = ''
      if (organizationObject.specific_topics) {
        tags = organizationObject.specific_topics.split(',').map(function (tag, i) {
          return (<a key={i} href='#'>{tag.trim()}</a>)
        })
      }
      if (organizationObject.communication_mode) {
        communication_modes = organizationObject.communication_mode.split(',').map(function (tag, i) {
          return (<a key={i} href='#'>{tag.trim()}</a>)
        })
      }
      let post_date = moment(organizationObject.post_date).format('LL')
      let updated_date = moment(organizationObject.updated_date).format('LL')
      let locale = this.props.intl.locale
      let author = organizationObject.authors[0]
      let first_author = (author && author.name) || 'unknown'
      let first_author_url = author ? '/' + locale + '/users/' + author.id : ''
      let last_author = organizationObject.authors.slice(-1)[0]
      let last_author_name = last_author.name
      let last_author_url = '/' + locale + '/users/' + last_author.id
      let awsUrl = process.env.REACT_APP_ASSETS_URL
      if (organizationObject.lead_image) {
        var comma = organizationObject.lead_image.search(',')
        var pic = awsUrl + encodeURIComponent(organizationObject.lead_image.url)
      }
      if (organizationObject.other_images.length) {
        var otherImg = awsUrl + encodeURIComponent(organizationObject.other_images[0].url)
      }

      return (
        <div>
          <div className='main-contents'>
            <Container className='detailed-case-component' fluid={true}>
              <Col md='3' className='sidepanel hidden-sm-down'>
                { organizationObject.location.country ?
                  <CountryMap city={organizationObject.location.city} countrycode={organizationObject.location.country} />
                  :
                  undefined
                }
                <p className='sub-heading'>
                  Keywords
                </p>
                <p className='sub-sub-heading'>
                  Tags:
                </p>
                <div className='tags'>
                {tags}
                </div>
                <p className='sub-sub-heading'>
                  Specific Topic(s):
                </p>
                <div className='tags'>
                {communication_modes}
                </div>
                <p className='sub-heading'>
                  Related Content
                </p>
                <div className='related-content'>
                  <a href='#'>Cases</a>
                  <a href='#'>Methods</a>
                  <a href='#'>Surveys</a>
                  <a href='#'>Datasets</a>
                </div>
              </Col>
              <Col md='8' xs='12' className='main-area'>
                <div className='case-box'>
                  <h2 className='category'>
                    Organization
                  </h2>
                  <h2 className='case-title'>
                    {organizationObject.title}
                  </h2>
                  { (pic && pic.length > awsUrl.length) ?
                    <div className='case-images'>
                      <img role='presentation' src={pic} />
                    </div>
                    :
                    (otherImg && otherImg.length > awsUrl.length) ?
                      <div className='case-images'>
                        <img role='presentation' src={otherImg} />
                      </div>
                    :
                    undefined
                  }
                  <div className='authorship-details'>
                    <p className='author-line'>
                      First submitted by&nbsp;
                      <a href={first_author_url}>
                        {first_author}
                      </a>
                    </p>
                    <p className='date-line'>
                    {post_date}
                    </p>
                    <p className='author-line'>
                      Most recent changes by&nbsp;
                      <a href='{last_author_url}'>
                        {last_author_name}
                      </a>
                    </p>
                    <p className='date-line'>
                    {updated_date}
                    </p>
                  </div>
                  <div className='case-html' dangerouslySetInnerHTML={{__html: organizationObject.body_en}} />
                </div>
              </Col>
              <Col md='1' className='case-tools hidden-sm-down'>
                <div className='top-icons'>
                  <a href='#'><img src={caseIconBookmark} alt='' /></a>
                  <a href='#'><img src={caseIconSettings} alt='' /></a>
                  <a href='#'><img src={caseIconFB} alt='' /></a>
                  <a href='#'><img src={caseIconTW} alt='' /></a>
                  <a href='#'><img src={caseIconShare} alt='' /></a>
                </div>
              </Col>
            </Container>
            <FloatingActionButton className='editButton'>
              <ContentPencil />
            </FloatingActionButton>
          </div>
        </div>)
    } else {
      return (<div></div>)
    }
  }
}

Organization.propTypes = {
  intl: intlShape.isRequired
}

export default injectIntl(Organization)
