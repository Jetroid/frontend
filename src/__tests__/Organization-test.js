import React from "react";
import renderer from "react-test-renderer";

import OrganizationDetails from "../components/OrganizationDetails";
import ItemDetails from "../components/ItemDetails/ItemDetails";
import { IntlProvider } from "react-intl";
import { getBestMatchingMessages } from "../utils/l10n";

import { MemoryRouter } from "react-router";
import { Organization } from "../containers/Organization";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import getMuiTheme from "material-ui/styles/getMuiTheme";
const muiTheme = getMuiTheme({});
import { mountWithIntl } from "../helpers/intl-enzyme-test-helper.js";
import data from "./org_data.json";
import intlProps from "../helpers/intl-props-test-helper.js";
import afterPromises from "../helpers/afterPromises";
import injectTapEventPlugin from "react-tap-event-plugin";
injectTapEventPlugin();

let fetchMock = require("fetch-mock");
jest.mock("material-ui/FloatingActionButton");
jest.mock("../components/BookmarkToggle");

fetchMock.get(
  process.env.REACT_APP_API_URL + "/organization/4219",
  JSON.stringify({ data: data })
);
fetchMock.get(
  process.env.REACT_APP_ASSETS_URL + "countries/fullname/Italy.svg",
  JSON.stringify({ data: data })
);

function setup() {
  const props = {
    intl: intlProps,
    location: { pathname: "/organization/4219" },
    match: { params: { nodeID: 4219 } },
    toggleFeatured: function() {},
    data: data
  };

  const enzymeWrapper = mountWithIntl(
    <MemoryRouter>
      <MuiThemeProvider muiTheme={muiTheme}>
        <Organization {...props} />
      </MuiThemeProvider>
    </MemoryRouter>
  );
  return {
    props,
    enzymeWrapper
  };
}

describe("containers", () => {
  describe("Organization", () => {
    it("should render proper data for org", done => {
      const { enzymeWrapper } = setup();
      afterPromises(done, () => {
        expect(enzymeWrapper.find(".sub-heading").length).toBe(1);
        expect(enzymeWrapper.find("h2.case-title").text()).not.toBe("");
      });
    });
  });
});

let props = setup().props;
let locale = "en-US";
props["details"] = OrganizationDetails;
let messages = getBestMatchingMessages(locale);

test("ItemDetails for Organization renders correctly", () => {
  const tree = renderer
    .create(
      <IntlProvider locale={locale} messages={messages}>
        <MemoryRouter><ItemDetails {...props} /></MemoryRouter>
      </IntlProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
