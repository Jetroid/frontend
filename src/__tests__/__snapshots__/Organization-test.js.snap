// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails for Organization renders correctly 1`] = `
<div>
  <div
    className="main-contents"
  >
    <div
      className="detailed-case-component container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="hidden-sm-down sidepanel hidden-sm-down col col-md-3"
        >
          <div />
          <div>
            <div
              className="case-map"
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "",
                  }
                }
              />
              <p
                className="case-location"
              >
                Italy
              </p>
            </div>
            <p
              className="sub-heading"
            >
              Keywords
            </p>
            <p
              className="sub-sub-heading"
            >
              Tags:
            </p>
            <div
              className="tags"
            >
              <a
                href="/search?tag=civic%20engagement"
                onClick={[Function]}
              >
                civic engagement
              </a>
              <a
                href="/search?tag=digital%20participation"
                onClick={[Function]}
              >
                digital participation
              </a>
              <a
                href="/search?tag=planning"
                onClick={[Function]}
              >
                planning
              </a>
              <a
                href="/search?tag=public"
                onClick={[Function]}
              >
                public
              </a>
            </div>
            <div />
          </div>
          <div />
        </div>
        <div
          className="main-area col-12 col-md-8"
        >
          <div
            className="case-box"
          >
            <h2
              className="category"
            />
            <h2
              className="case-title"
            >
              Urban Interactive Studio
            </h2>
            <ul
              className="icons-mobile clearfix hidden-sm-up"
            >
              <li />
              <li>
                <div
                  className="SocialMediaShareButton SocialMediaShareButton--facebook"
                  onClick={[Function]}
                  style={Object {}}
                >
                  <img
                    alt=""
                    src="pp-case-icon-fb.svg"
                  />
                </div>
              </li>
              <li>
                <div
                  className="SocialMediaShareButton SocialMediaShareButton--twitter"
                  onClick={[Function]}
                  style={Object {}}
                >
                  <img
                    alt=""
                    src="pp-case-icon-tw.svg"
                  />
                </div>
              </li>
            </ul>
            <div
              className="single-image"
            >
              <img
                alt={undefined}
                className="portrait"
                sizes={undefined}
                src="http://assets.participedia.xyz.s3-website-us-east-1.amazonaws.com/uis_favicon_0.png"
                srcSet={undefined}
              />
            </div>
            <div
              className="authorship-details"
            >
              <p
                className="author-line"
              >
                First submitted by 
                <a
                  href="/users/undefined"
                  onClick={[Function]}
                >
                  ecrespin
                </a>
              </p>
              <p
                className="date-line"
              >
                <TimeAgo />
              </p>
              <p
                className="author-line"
              >
                Most recent changes by 
                <a
                  href="/users/undefined"
                  onClick={[Function]}
                >
                  ecrespin
                </a>
              </p>
              <p
                className="date-line"
              >
                <TimeAgo />
              </p>
            </div>
            <div
              className="case-html"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<h3>Mission and Purpose</h3><p><a href=\\"http://www.urbaninteractivestudio.com\\">Urban Interactive Studio</a> believes that communities should be shaped by people with the assistance of experts, rather than handed down as a finished product. The UIS mission is to give people a voice in shaping the future of their communities.</p><h3>Founder and Philosophy</h3><p>Chris Haller, the CEO and Founder of Urban Interactive Studio, is a nationally-recognized User Experience designer and Online Engagement strategist with a broad background in local government, urban and regional planning and communication technologies. In April 2011 he was named one of Planetizen's Top 25 thinkers in Urban Planning Technology, and he brings a sense of creativity and curiosity to all of UIS's clients and projects.</p><p>UIS believes that people and their diverse perspectives are the key to livable communities. But growing use of Web and mobile technology means that traditional methods of public participation like in-person meetings need to be supplemented with digital activities. UIS makes these digital activites intuitive so that complex information is easily accessible to citizens and they are in the driver’s seat as they explore and provide feedback on options related the future of their community.</p><p>Urban Interactive Studio also publishes <a href=\\"http://www.engagingcities.com\\">EngagingCities</a>, the prominent online magazine that shares creative strategies and new technologies to foster public engagement for livable communities.</p><h3>Specializations, Tools and Projects</h3><p>Urban Interactive Studio is a cross-disciplinary team of planners, designers and web developers that specializes in providing <strong>intuitive solutions for digital public engagement</strong>. We’re a small but steadily growing firm at the cutting edge of public process, helping local governments, elected officials, planning and engineering firms, and nonprofits educate, inform and learn from citizens and stakeholders.</p><p>UIS's flagship product,<a href=\\"http://www.urbaninteractivestudio.com/engagingplans\\"> EngagingPlans</a>, is an interactive public engagement platform that enables project teams to facilitate broad-scale public outreach and participation. The EngagingPlans Core includes project updates, event calendars and timelines, a searchable document library, FAQs, a survey builder, polls and open questions.</p><p>The <a href=\\"http://www.urbaninteractivestudio.com/appsuite\\">EngagingPlans App Suite</a> extends Core functionality with a range of apps that offer unique, intuitive opportunities for public involvement and feedback. From community mapping to interactive plan documents, these apps can be tailored to meet the engagement objectives of each project phase.</p><p>UIS has worked with dozens of clients across the country and globe to help connect experts and civic leaders with the local insights necessary to build the best communities possible. Many <a href=\\"http://www.urbaninteractivestudio.com/blog\\">past project stories</a> are chronicled on the blog.</p><h3>External Links</h3><p><a href=\\"http://www.urbaninteractivestudio.com\\">http://www.urbaninteractivestudio.com</a></p><p><a href=\\"http://twitter.com/urbaninteractiv\\">@urbaninteractiv</a></p><p><a href=\\"http://twitter.com/challer\\">@challer</a></p><p><a href=\\"http://www.engagingcities.com\\">http://www.engagingcities.com</a></p><p><a href=\\"http://twitter.com/engagingcities\\">@engagingcities</a></p>",
                }
              }
            />
          </div>
        </div>
        <div
          className="case-tools hidden-sm-down col col-md-1"
        >
          <div
            className="top-icons"
          >
            <div
              className="SocialMediaShareButton SocialMediaShareButton--facebook"
              onClick={[Function]}
              style={Object {}}
            >
              <img
                alt=""
                src="pp-case-icon-fb.svg"
              />
            </div>
            <div
              className="SocialMediaShareButton SocialMediaShareButton--twitter"
              onClick={[Function]}
              style={Object {}}
            >
              <img
                alt=""
                src="pp-case-icon-tw.svg"
              />
            </div>
            <div
              className="SocialMediaShareButton SocialMediaShareButton--linkedin"
              onClick={[Function]}
              style={Object {}}
            >
              <img
                alt=""
                src="pp-case-icon-ln.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <a
      href="/organization/undefined/edit"
      onClick={[Function]}
    />
  </div>
</div>
`;
